pages/index.js

```diff
@@ -4,12 +4,17 @@ import useSWR from "swr";
 import styled from "styled-components";
 import useLocalStorageState from "use-local-storage-state";
 import ShoppingPurchasedItem from "@/components/ShoppingPurchasedItem";
+import ShoppingCategoryFilter from "@/components/ShoppingCategoryFilter";
 export default function HomePage() {
   // Mark as purchased
   const [purchasedIds, setPurchasedIds] = useLocalStorageState(
     "purchased-items",
     { defaultValue: [] }
   );
+  const [selectedCategory, setSelectedCategory] = useLocalStorageState(
+    "category-items",
+    { defaultValue: null }
+  );
```

Comment:
```suggestion
  const [selectedCategory, setSelectedCategory] = useState(null);
```
Why are you using localStorage for the filtering? If it makes sense to you, keep it. But I don't think it's needed personally...